src/score_explain.c

```diff
@@ -1,17 +1,19 @@
 #include "score_explain.h"
 #include "rmalloc.h"
+#include "config.h"
 
-static void recExplainReply(RedisModuleCtx *ctx, RSScoreExplain *scrExp) {
+static void recExplainReply(RedisModuleCtx *ctx, RSScoreExplain *scrExp, int depth) {
   int numChildren = scrExp->numChildren;
 
-  if (numChildren == 0) {
+  if (numChildren == 0 ||
+     (depth == REDIS_ARRAY_LIMIT && !isFeatureSupported(NO_REPLY_DEPTH_LIMIT))) {
```

Comment:
As we discussed, since we increment the depth by 2 (add an array within an array),
We should check depth >= REDIS_ARRAY_LIMIT - 1

And Codecov is complaining on this block